test/index.test.ts

```diff
@@ -208,6 +209,103 @@ describe('PaymentSocket', function () {
     })
   })
 
+  describe('pay', function () {
+    it('should lower the maximum balance and resolve when the balance is below the maximum', async function () {
+      await this.clientSocket.pay(2000)
+      assert.equal(this.clientSocket.balance, '-2000')
+
+      await this.clientSocket.pay(1500)
+      assert.equal(this.clientSocket.balance, '-3500')
+    })
+
+    it('should reject if the amount is higher than the other party will accept', async function () {
+      this.serverSocket.setMaxBalance(900)
+      let errored = false
+      try {
+        await this.clientSocket.pay(2000)
+      } catch (err) {
+        errored = true
+      }
+      assert.equal(errored, true)
+    })
+
+    it('should resolve when the socket has stabilized, even if the user changed the limits before it did', async function () {
```

Comment:
What's the rationale for this behavior? If the pay call is resolving, wouldn't a user expect it to have paid the specified amount?

Reply:
The problem is that if you lower the limit the promise will never resolve. We could handle the case where you raise it more differently, but that seems a bit strange too. Basically if you have multiple things calling the relative methods at once before the first one is done there's going to be some strange behavior